apps/test/feedbackTests.js

```diff
@@ -13,6 +13,103 @@ global.window = {};
 global.document = {};
 
 /**
+ * Loads blocks into the workspace, then calls
+ * checkForEmptyContainerBlockFailure_ and validates
+ * that the result matches the expected result.
+ */
+describe("checkForEmptyContainerBlockFailure_", function () {
+  var studioApp;
+  var TestResults;
+
+  // create our environment
+  beforeEach(function () {
+    testUtils.setupTestBlockly();
+    studioApp = testUtils.getStudioAppSingleton();
+    TestResults = studioApp.TestResults;
+  });
+
+  var checkResultForBlocks = function (args) {
+    studioApp.loadBlocks(args.blockXml);
+
+    // make sure we loaded correctly. text wont match exactly, but make sure if
+    // we had xml, we loaded something
+    var loaded = Blockly.Xml.domToText(
+        Blockly.Xml.blockSpaceToDom(Blockly.mainBlockSpace));
+    assert(!args.blockXml || loaded, "either we didnt have  input xml" +
+        "or we did, and we loaded something");
+
+    assert.equal(args.result,
+        studioApp.feedback_.checkForEmptyContainerBlockFailure_());
+  };
+
+  it("returns ALL_PASS when no blocks are present", function () {
+    checkResultForBlocks({
+      result: TestResults.ALL_PASS,
+      blockXml: ''
+    });
+  });
+
+  it ("returns ALL_PASS when no container blocks are present", function () {
+    checkResultForBlocks({
+      result: TestResults.ALL_PASS,
+      blockXml: '<xml><block type="text_print"></block></xml>'
```

Comment:
No need to convert this, but FYI going forward there are a few utility functions in block_utils that make some of this xml generation easier (you can do stuff like blockUtils.blockOfType('textPrint'))